crates/core/src/token.rs

```diff
@@ -870,7 +800,7 @@ pub enum AmountParseError {
         "Error decoding token amount, too many decimal places: {0}. Maximum \
          {1}"
     )]
-    ScaleTooLarge(u32, u8),
+    ScaleTooLarge(usize, u8),
```

Comment:
should we avoid this in light of #826?

Reply:
it's ok in error type - it doesn't affect the state machine